dot/rpc/modules/grandpa.go

```diff
@@ -71,3 +96,84 @@ func (gm *GrandpaModule) ProveFinality(r *http.Request, req *ProveFinalityReques
 
 	return nil
 }
+
+// RoundState returns the state of the current best round state as well as the ongoing background rounds.
+func (gm *GrandpaModule) RoundState(r *http.Request, req *EmptyRequest, res *RoundStateResponse) error {
+	voters := gm.blockFinalityAPI.GetVoters()
+	votersPkBytes := make([]ed25519.PublicKeyBytes, len(voters))
+	for i, v := range voters {
+		votersPkBytes[i] = v.PublicKeyBytes()
+	}
+
+	prevotes, pvEquivocations := gm.blockFinalityAPI.PreVotes()
+	precommits, pcEquivocations := gm.blockFinalityAPI.PreCommits()
+
+	missingPrevotes, err := toAddress(difference(votersPkBytes, prevotes))
+	if err != nil {
+		return err
+	}
+
+	missingPrecommits, err := toAddress(difference(votersPkBytes, precommits))
+	if err != nil {
+		return err
+	}
```

Comment:
equivocations should be counted as votes, since `difference` returns authorities that haven't voted. I'd suggest updating `PreVotes()` and `PreCommits()` to return just one list  `[]ed25519.PublicKeyBytes` that contains all the voters that are in both

Reply:
Done!